tests/test_render.py

```diff
@@ -0,0 +1,29 @@
+# -- encoding: utf-8 --
+from __future__ import unicode_literals
+
+import pytest
+
+
+@pytest.mark.parametrize('locale', ('en', 'fi', 'sv', 'pt-BR'))
+def test_babel_render(client, locale):
+    """
+    Test the middleware and the rendery bits.
+    """
+    response = client.get('/', HTTP_ACCEPT_LANGUAGE=locale)
+    # "Parse" the key-value format
+    content = dict(
+        kv.split('=', 1)
+        for kv
+        in response.content.decode('utf-8').strip().splitlines()
+    )
```

Comment:
Since `kv.split('=', 1)` is guaranteed to return a 2-tuple (unless the line doesn't have a `=`, which is a bug in the test case anyway), and `dict()` accepts an iterable of 2-tuples, doing it like this saves the identity dictcomp. However sure, I can uninline the decode-strip-split.